### DISCUSSION TITLE
encode the file data?

### DISCUSSION THREAD

**Discussion by whatatimetobealive2022** [obrassard/shc-extractor#32] [Q&A]
Obviously you can just save the SHC qr and have multiple copies of it. but what im curious about, if you scan the shc qr code, and you now have the SHC:1/2................................................. long string of numbers.  is there a program or site where you input the numeric data, and it converts back into the original QR code image. not looking to alter any info, just curious to generate the QR code with only the numeric data.

**Comment by obrassard** [comment]
Hello ! There is a collection of Jupyter Notebooks demonstrating the creation of SMART Health Cards in different programming languages here https://github.com/dvci/health-cards-walkthrough

**Comment by timlegge** [comment]
any numbe of barcode generating sites can generate the exact QR core from
the same numeric data.  The qr code is simply a representation of the data

**Reply by whatatimetobealive2022** [reply]
i have been trying that, but the end resulting QR code cannot be verified by the issuer.

in the barcode generating sites, I dont have an SHC option, i just copy and paste it as a Text file. then convert. it looks like a normal QR code, but it wont scan by the issuer that the original image was made from.

**Reply by timlegge** [reply]
be sure that you include the shc:/

I just grabbed a sample and pasted it into qrcode-monkey.com and it validated fine when I checked it.  It has to be exactly the same data as was read from the valid QR code.  That is, it needs to be valid

**Reply by whatatimetobealive2022** [reply]
yes, the file is shc:/5676.....................

**Reply by timlegge** [reply]
again, if you have the data from a valid smart health card, that is one that you are able to validate - simply taking the exact shc:/... data and generating a QR code will produce a valid QR code.  Confirm when you scan the code that the data from both qr codes is identical

**Comment by whatatimetobealive2022** [comment]
i got it to work. you were right, the full info needs to be matching. i was using a bad scan, and didnt have all characters. i rescanned it, and got more data, and it worked fine. thanks
